### ISSUE TITLE
i18n for non latin languages

### ISSUE THREAD

**Issue by Joxit** [pelias/pelias#914]
Hi there,

Last year I worked on [language management on maps](https://blog.jawg.io/embrace-streets-v2/). Today I wanted to check the language management in Pelias and I think we could improve some things for non-latin languages.

### Use-cases

I am traveling to a country that does not use Latin characters. I know a lot of features/documents are translated into English, but not available in my native language. For example [慶應義塾大学病院/Keio University Hospital](https://www.openstreetmap.org/way/509241418) has a translation in English, Spanish and Korean.

When I use Pelias for 慶應義塾大学病院 in English the result is pretty good:
[/v1/search?lang=en&text=慶應義塾大学病院](https://pelias.github.io/compare/#/v1/search?lang=en&text=%E6%85%B6%E6%87%89%E7%BE%A9%E5%A1%BE%E5%A4%A7%E5%AD%A6%E7%97%85%E9%99%A2)
```
0) Keio University Hospital, Tokyo, Japan
```

But when I use Pelias for 慶應義塾大学病院 in Franch, the result is in Japanese.
[/v1/search?lang=fr&text=慶應義塾大学病院](https://pelias.github.io/compare/#/v1/search?lang=fr&text=%E6%85%B6%E6%87%89%E7%BE%A9%E5%A1%BE%E5%A4%A7%E5%AD%A6%E7%97%85%E9%99%A2)
```
0) 慶應義塾大学病院, Tokyo, Japon
```

I also know that in OSM, when the name is not available in English, we can use some variants. This could be helpful to improve the names stored in Pelias.

### Attempted Solutions

The naive solution would be to use the English name when the default name is not Latin. But that won't work, because a Japanese would have the name in Latin as well ....

**Comment by missinglink** [comment]
Yeah the existing policy is to serve your requested language where available, otherwise show the 'default' name, which could be in any language really, but for the most part its one of the official local languages.

I feel like the preference for what to show is somewhat subjective, what about French people who don't speak English? It would be odd for them to see English labels for Japanese places, even more so if they speak some Japanese.

In your case since you are multilingual we iterate over the values in the `Accept-Language` header until we find one which matches rather than only using the highest priority language.

Out of curiosity, what `Accept-Language` header does your browser send?

**Comment by Joxit** [comment]
Hum... You're right, this will always be subjective. I have the example of Morocco in mind which has Arabic as its native language but the main foreign language is French .... French is often seen in the streets and for communication with officials.... So English would be a really strange fallback when the name is written in Arabic. :thinking: 

I like the idea of iterate over `Accept-Language` header :+1: 

My `Accept-Language` header is `fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7`

**Comment by missinglink** [comment]
I can imagine some labels would be rendered in a mix of French, English and Japanese, and potentially more depending on the `Accept-Language` string.

In a worst case scenario every component of the label would be in a different language 😂, but that's more of a data issue than a code one.

We've recently introduced some HTTP request caching and found the `Accept-Language` header was difficult thing to generate hashes for since it's not clear which language is going to be used until after the API has responded. I guess in this case we could simplify the header and use that as the key.
